File: leetcode/tree/same_tree.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isSameTree(p *TreeNode, q *TreeNode) bool {
	if p == nil && q == nil {
		return true
	}

	if p != nil && q != nil {
		if p.Val != q.Val {
			return false
		} else {
			return isSameTree(p.Left, q.Left) && isSameTree(p.Right, q.Right)
		}
	}

	return false
}

func main() {
	p := &TreeNode{3, nil, nil}
	p.Left = &TreeNode{1, nil, nil}
	p.Right = &TreeNode{4, nil, nil}
	q := &TreeNode{3, nil, nil}
	q.Left = &TreeNode{1, nil, nil}
	q.Right = &TreeNode{2, nil, nil}
	if isSameTree(p, q) {
		fmt.Println("Same")
	} else {
		fmt.Println("Not same")
	}
}

File: leetcode/tree/same_tree.py
from common import TreeNode
class Solution:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        if not p and not q:
            return True
        if p and q:
            if p.val != q.val:
                return False
            return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)
        return False

if __name__ == "__main__":
    q = TreeNode(3)
    q.left = TreeNode(4)
    q.left = TreeNode(5)
    p = TreeNode(3)
    p.left = TreeNode(4)
    p.left = TreeNode(5)
    if Solution().isSameTree(q, p):
        print("Same")
    else:
        print("Not Same")